scalalib/src/mill/scalalib/SemanticDbJavaModule.scala

```diff
@@ -84,7 +84,6 @@ trait SemanticDbJavaModule extends JavaModule { hostModule =>
             }
         )
           .filterNot(_ == "-Xfatal-warnings")
-        T.log.debug(s"effective scalac options: ${scalacOptions}")
```

Comment:
The idea is, that a user only sees this when option `--debug` is enabled. I agree, that we probably used the wrong mechanism to transport this to a BSP client, but we should not be forced to avoid debug logs in favor of a nice IDE experience. Why is the BSP client showing debug messages without the user explicitly asking for it? Isn't there any other option?

Reply:
With the way the logger was set up before we were doing exactly that, telling the build client **show this to the user**. By switching it to `onBuildLogMessage` we avoid this but these are still sent to the client and the client will log this out every time. I looked to see if there was a `MessageType.DEBUG` which would be perfect here, but there really isn't that. So you end up in a situation where if you would look in the client log you'll just see this spammed in there since it's coming as an info message. This _sort_ of seems like a missing feature of BSP.

Reply:
Ok. I already have a stashed patch which enabled `--debug` in the BSP server, when it was enabled at`BSP/install`-time. So, if we only send this to the client when `--debug` is enabled (`Logger.debugEnabled` is `true`), we should be fine. WDYT?

Reply:
Yea I think that should be fine, and useful.

Reply:
Ok. I already have a stashed patch which enabled `--debug` in the BSP server, when it was enabled at`BSP/install`-time. So, if we only send this to the client when `--debug` is enabled (`Logger.debugEnabled` is `true`), we should be fine. WDYT?

It's basically this: github.com/com-lihaoyi/mill/pull/1977/commits/3328d99513416a9535e739a4d809d2078f967fdb

Reply:
* Created PR #2007

Reply:
Ah ok great. Should we merge that other pr, and then I can rebase and updaet?

Reply:
The other PR is now independent of this one. Just keep the debug log message.